.github/workflows/test.yml

```
@@ -24,14 +26,25 @@ jobs:
             - name: Setup Job and Install Dependencies
               uses: ./.github/actions/setup-job
 
+            - name: Extract branch name
+              shell: bash
+              run: echo "BRANCH=$(npx slugify-cli ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}})" >> $GITHUB_OUTPUT
+              id: extract_branch
+
+            - name: Extract doc preview url
+              run: echo "DOC_PREVIEW_URL=https://${{ steps.extract_branch.outputs.branch }}--spectrumwc.netlify.app/" >> $GITHUB_OUTPUT
```

You can put this directly in a `GITHUB_ENV` instead and that will save you the steps of redefining the output as a local `ENV` in each step you need to access it. Same with the `BRANCH` value above.
Putting it as a global env variable means it's accessible in all scripts, not just in the dependent steps.

```suggestion
              run: echo "DOC_PREVIEW_URL=https://${{ steps.extract_branch.outputs.branch }}--spectrumwc.netlify.app/" >> $GITHUB_ENV
```